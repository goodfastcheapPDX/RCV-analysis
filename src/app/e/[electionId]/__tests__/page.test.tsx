import { render } from "@testing-library/react";
import { notFound } from "next/navigation";
import { describe, expect, it, vi } from "vitest";
import ElectionPage from "../page";

describe("Election Page", () => {
  it("should match snapshot for valid election with test data", async () => {
    const component = await ElectionPage({
      params: Promise.resolve({ electionId: "portland-20241105-gen" }),
    });
    const { container } = render(component);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should call notFound for invalid election", async () => {
    await expect(
      ElectionPage({
        params: Promise.resolve({ electionId: "invalid-election" }),
      }),
    ).rejects.toThrow("NEXT_HTTP_ERROR_FALLBACK;404");
  });
});
