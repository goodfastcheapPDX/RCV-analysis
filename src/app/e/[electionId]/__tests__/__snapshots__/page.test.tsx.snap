// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Election Page > should match snapshot for valid election with test data 1`] = `
<div
  class="space-y-6"
>
  <div>
    <h1
      class="text-3xl font-bold"
    >
      Portland General Election 2024
    </h1>
    <p
      class="-foreground mt-2"
    >
      Analyze contests within this election
    </p>
  </div>
  <div
    class="grid gap-4"
  >
    <h2
      class="text-xl font-semibold"
    >
      Contests
    </h2>
    <div
      class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm hover:shadow-md transition-shadow"
      data-slot="card"
    >
      <div
        class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6"
        data-slot="card-header"
      >
        <div
          class="leading-none font-semibold"
          data-slot="card-title"
        >
          <a
            class="hover:underline"
            href="/e/portland-20241105-gen/c/d2-3seat"
          >
            City Council District 2 (3 seats)
          </a>
        </div>
        <div
          class="-foreground text-sm"
          data-slot="card-description"
        >
          <span
            class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden text-foreground [a&]:hover:bg-accent [a&]:hover:text-accent-foreground"
            data-slot="badge"
          >
            3
             seat
            s
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
